File: apps/agent/app/deps.py
"""FastAPI dependency injection providers."""

from fastapi import Depends
from app.config import settings
from app.llm.base import LLMProvider
from app.llm.factory import get_llm_provider
from app.session.base import SessionStore
from app.session.factory import get_session_store
from app.agent.runner import AgentRunner


def get_llm() -> LLMProvider:
    """Provide configured LLM provider."""
    return get_llm_provider(settings)


def get_session_store_dep() -> SessionStore:
    """Provide configured Session store."""
    return get_session_store(settings)


def get_agent_runner(
    llm: LLMProvider = Depends(get_llm),
    session_store: SessionStore = Depends(get_session_store_dep),
) -> AgentRunner:
    """Provide initialized AgentRunner."""
    return AgentRunner(llm=llm, session_store=session_store)

File: apps/agent/src/deps.ts
import { config } from './config';
import { getLLMProvider } from './llm/factory';
import { getSessionStore } from './session/factory';
import { AgentRunner } from './agent/runner';
import type { LLMProvider } from './llm/base';
import type { SessionStore } from './session/base';

let defaultRunner: AgentRunner | null = null;

export function getAgentRunner(
  customLlm?: LLMProvider,
  customSessionStore?: SessionStore
): AgentRunner {
  if (customLlm || customSessionStore) {
    const llm = customLlm ?? getLLMProvider();
    const store = customSessionStore ?? getSessionStore();
    return new AgentRunner(llm, store);
  }

  if (!defaultRunner) {
    const llm = getLLMProvider(config.LLM_PROVIDER);
    const store = getSessionStore(config.SESSION_STORE);
    defaultRunner = new AgentRunner(llm, store);
  }

  return defaultRunner;
}
